changeset.go

```diff
@@ -1338,28 +1382,27 @@ func (cs *Changeset) withUpsertOp(ctx context.Context, tr *ChangesetResource, ol
 		item.From = string(from)
 		item.UID = string(old.GetUID())
 	}
-	tr.Spec.Items = append(tr.Spec.Items, item)
-	tr, err = cs.update(tr)
+	ur.resource.Spec.Items = append(ur.resource.Spec.Items, item)
+	ur.resource, err = cs.update(ur.resource)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 	if err := fn(); err != nil {
 		return nil, trace.Wrap(err)
 	}
-	tr.Spec.Items[len(tr.Spec.Items)-1].Status = OpStatusCompleted
-	return cs.update(tr)
+	ur.resource.Spec.Items[len(ur.resource.Spec.Items)-1].Status = OpStatusCompleted
```

Comment:
isn't this the same as setting `item.Status = OpStatusCompleted` before appending it? Seems easier that way